catalogs/sources/gtfs/schedule/us-california-santa-maria-regional-transit-gtfs-26.json

```diff
@@ -15,7 +15,7 @@
         }
     },
     "urls": {
-        "direct_download": "http://data.trilliumtransit.com/gtfs/smat-ca-us/smat-ca-us.zip",
+        "direct_download": "",
```

Comment:
1) Currently, changing the name of a file would also trigger changing the direct download URL and break pipeline workflows. We could change the direct download URL structure so it’s non-descriptive (only the source ID for instance), or discuss other potential solutions.

2) A direct download URL needs to be available for the source to be valid. I’m assuming there’s no available replacement since you left it empty? We have talked about adding feed_end_date to the schema to make it easier to see if the most recently available URL is still in use.

Reply:
1. This is unfortunate. So what happens when an agency changes their name?  When a URL changes what they represent?   If this is the case, I would suggest only having the ID and/or geography (something that isn't going to change) in the name.

2. That is correct.

Reply:
1. Agreed. [Issue created here. ](https://github.com/MobilityData/mobility-database-catalogs/issues/134)

2. The schema states that the URL should automatically open the source information, and doesn't describe anything regarding data recency. In instances where the URL is not up-to-date but there is no known available alternative, we've still been including it in the catalogs, as this is a very common occurrence and deleting the source entirely seemed counterproductive for data access. Once [historical URLs](https://github.com/MobilityData/mobility-database-catalogs/issues/136) are implemented in the schema, we could include these URLs there instead for clarity.

Reply:
> 1. Currently, changing the name of a file would also trigger changing the direct download URL and break pipeline workflows. We could change the direct download URL structure so it’s non-descriptive (only the source ID for instance), or discuss other potential solutions.

Here, it's the latest dataset URL that is affected, not the direct download one (so it's system generated not provided by the user), but I agree that in any cases it should be stable enough that changing the filename won't break the pipeline.

> 2. A direct download URL needs to be available for the source to be valid. I’m assuming there’s no available replacement since you left it empty? We have talked about adding feed_end_date to the schema to make it easier to see if the most recently available URL is still in use.

I think adding feed_end_date and feed_start_date would be a good idea. It would allow us not to update the latest url if the feed expiration date is reached. Also, we could always keep the direct download url after adding the historical data.